girder/models/model_base.py

```diff
@@ -907,9 +1038,57 @@ def setUserAccess(self, doc, user, level, save=False):
                      Set this to False if you want to wait to save the
                      document for performance reasons.
         :type save: bool
+        :param flags: List of access flags to grant to the group.
+        :type flags: specific flag identifier, or a list/tuple/set of them
+        :param currentUser: The user performing this action. Only required if attempting
+            to set admin-only flags on the resource.
         :returns: The modified resource document.
         """
-        return self._setAccess(doc, user['_id'], 'users', level, save)
+        return self._setAccess(doc, user['_id'], 'users', level, save, flags, currentUser)
+
+    def hasAccessFlags(self, doc, user=None, flags=None):
+        """
+        Test whether a specific user has a given set of access flags on
+        the given resource. Returns True only if the user has all of the
+        flags by virtue of either group membership, public flags, or
+        explicit access granted to the user.
+
+        :param doc: The resource to test access on.
+        :type doc: dict
+        :param user: The user to check against.
+        :type user: dict or None
+        :param flags: A flag or set of flags to test.
+        """
+        if user and user['admin']:
+            # Short-circuit the case of admins
+            return True
+
+        if not flags:
+            # Special case if no flags are passed
+            return True
+
+        if not isinstance(flags, (list, tuple, set)):
+            flags = {flags}
```

Comment:
here `flags` is a set, in other similar places it is a list and a tuple.  same nit as above, though maybe this one has a better claim to be made a set.